src/IoC/IoC.cs

```diff
@@ -0,0 +1,31 @@
+using Services;
+using Controllers;
+namespace IoC;
+
+public class Ioc
+{
+    private static Ioc? _instance;
+    private static Dictionary<string, object> _serviceInstances = new Dictionary<string, object>();
+    
+    private Ioc()
+    {
+        _serviceInstances.Add("ProductRepository", new ProductRepository());
```

Comment:
This is not meant to only register revices. Maybe use a more generic name like `_iocInstanceDict` or `_iocInstanceMap`

Reply:
Understood!